.github/workflows/ci.yml

```diff
@@ -206,7 +206,7 @@ jobs:
     strategy:
       matrix:
         platform: [amd64, arm64]
-        python_version: ['3.8', '3.11']
+        python_version: ['3.9', '3.11']
```

Comment:
While working on the new features, we realized that some tests and functionality didn't work on Python 3.8 (things like `list[float]`, `graphlib`), and we didn't catch it until deployment phase. So, this change was made to ensure we check on the lowest supported version...

Reply:
OK, I generated https://github.com/NVIDIA/cuda-quantum/issues/1313 so that we can consider changing the branch protection rules and the CI to go back to `3.8` once this is merged.